xfields/ibs/_equilibrium_emittance.py

```diff
@@ -334,44 +374,60 @@ def compute_emittance_evolution(
             current_emittances[1] = forced_emittance_y
 
         # Append current values to lists
-        time.append(time_step)
-        emittances_x_list.append(current_emittances[0])
-        emittances_y_list.append(current_emittances[1])
-        emittances_z_list.append(current_emittances[2])
+        time_deltas.append(time_step)
+        res_gemitt_x.append(current_emittances[0])
+        res_gemitt_y.append(current_emittances[1])
+        res_gemitt_zeta.append(current_emittances[2])
         T_x.append(ibs_growth_rates[0])
         T_y.append(ibs_growth_rates[1])
         T_z.append(ibs_growth_rates[2])
 
-        # Compute tolerance
-        if it > 0:
-            tol = np.max(np.abs((current_emittances - previous_emittances) / previous_emittances))
+        # Compute tolerance (but not at first step since there is no previous value)
+        if iterations > 0:
+            tolerance = np.max(np.abs((current_emittances - previous_emittances) / previous_emittances))
+
         # Store current emittances for the next iteration
         previous_emittances = current_emittances.copy()
 
         # Update time step for the next iteration
         time_step = 0.01 / np.max((ibs_growth_rates, twiss.damping_constants_s))
 
-        it += 1
-
+        iterations += 1
+    # ----------------------------------------------------------------------------------------------
+    # We have exited the loop, we have converged. Construct a Table with the results and return it
     print("\nConverged!")
-    return (
-        np.cumsum(time),
-        emittances_x_list,
-        emittances_y_list,
-        emittances_z_list,
-        T_x,
-        T_y,
-        T_z,
-    )
-    # return Table(
+    # result_table = Table(
     #     data={
     #         "time": np.cumsum(time),
     #         "gemitt_x": emittances_x_list,
     #         "gemitt_y": emittances_y_list,
-    #         "gemitt_z": emittances_z_list,
+    #         "gemitt_zeta": emittances_z_list,
     #         "Tx": Tx,
     #         "Ty": Ty,
     #         "Tz": Tz,
     #     },
     #     index="time",
     # )
+    # Provide global quantities as well
+    # result_table._data.update(
+    #     {
+    #         "damping_constants_s": twiss.damping_constants_s,
+    #         "partition_numbers": twiss.partition_numbers,
+    #         "eq_gemitt_x": emittances_x_list[-1],
+    #         "eq_gemitt_x": emittances_y_list[-1],
+    #         "eq_gemitt_x": emittances_z_list[-1],
+    #         "sr_eq_gemitt_x": twiss.eq_gemitt_x,
+    #         "sr_eq_gemitt_y": twiss.eq_gemitt_y,
+    #         "eq_nemitt_zeta": twiss.eq_nemitt_zeta,
+    #     }
+    # )
+    # return result_table
+    return (
+        np.cumsum(time_deltas),
+        res_gemitt_x,
+        res_gemitt_y,
+        res_gemitt_zeta,
+        T_x,
+        T_y,
+        T_z,
+    )
```

Comment:
Nothing changed here, just commented out code for when we decide to return an `xtrack.Table`.